src/bp/lang-server/service/index.ts

```diff
@@ -45,75 +54,172 @@ export default class LanguageService {
     return this._ready
   }
 
-  listModels(): AvailableModel[] {
-    return _.values(this._models)
+  listFastTextModels(): AvailableModel[] {
+    const allModels = _.values(this._models)
+    return allModels.map(m => m.fastTextModel)
   }
 
   private _discoverModels() {
     if (!fs.existsSync(this.langDir)) {
-      return []
+      return
     }
 
     const files = fs.readdirSync(this.langDir)
-    return files.forEach(f => {
+
+    type modelFileInfo = {
+      domain: string
+      langCode: string
+      file: string
+      dim?: number
+    }
+    const filesInfo: modelFileInfo[] = []
+
+    files.forEach(f => {
       // Examples:  "scope.en.300.bin" "bp.fr.150.bin"
-      const regex = /(\w+)\.(\w+)\.(\d+)\.bin/i
-      const match = f.match(regex)
-      if (!match) {
+      const fastTextModelsRegex = /(\w+)\.(\w+)\.(\d+)\.bin/i
```

Comment:
Maybe just a personal preference, but it's nice to have regex defined as constants outside the class (at the top of it). 
eg: const FASTTEXT_MODEL_REGEX = ...

Reply:
+ we gain just a little computation time as the regex instance doesn't get rebuilt for each files